File: 25/a.java
public class a {

    public static void main(String[] args) {
        java.util.Scanner sc = new java.util.Scanner(System.in);
        int n = sc.nextInt();
        //int[] a = new int[n];
        int odd = 0, even = 0, lastOdd = 0, lastEven = 0;

        for (int i = 0; i < n; i++) {
            int val = sc.nextInt();

            if (val % 2 == 0) {
                even++;
                lastEven = i;
            } else {
                odd++;
                lastOdd = i;
            }
        }

        sc.close();
        System.out.println(odd > even ? (lastEven + 1) : (lastOdd + 1));
    }
}

File: 25/a.cpp
#include <cstdio>

int main() {
  int n, x, odd = 0, even = 0, last_odd = -1, last_even = -1;
  scanf("%d", &n);

  for (int i = 0; i < n; ++i) {
    scanf("%d", &x);

    if (x % 2 == 0) {
      ++even, last_even = i + 1;
    } else {
      ++odd, last_odd = i + 1;
    }
  }

  printf("%d\n", odd < even ? last_odd : last_even);
  return 0;
}
